iOS Client/ArticlesController/ArticlesCoordinator.swift

```diff
@@ -49,21 +49,6 @@ class ArticlesCoordinator: NSObject, Coordinator, StoreSubscriber, UINavigationC
         build(newRoute: newRoute)
         route = newRoute
         
-        let articlesSafariOverlay: URL? = state.navigationState.safariOverlays[.articles]!
-        if self.overlay == nil, let overlay = articlesSafariOverlay {
-            self.overlay = overlay
-            let overlayVC = SFSafariViewController(url: overlay)
-            overlayVC.delegate = self
-            self.navigationController.present(overlayVC, animated: true, completion: nil)
-        } else if self.overlay != nil, articlesSafariOverlay == nil {
-            self.overlay = nil
-            self.navigationController.dismiss(animated: true, completion: nil)
-        }
-    }
-    
-    // MARK: - SafariViewControllerDelegate
-    func safariViewControllerDidFinish(_ controller: SFSafariViewController) {
-        MBStore.sharedStore.dispatch(SelectedArticleLink(url: nil))
-        self.overlay = nil
+        self.displaySafariVC(forURL: state.navigationState.safariOverlays[.articles]!)
```

Comment:
How do you know that this is not going to be nil?

Reply:
I invite it to be `nil`. The method will pop that overlay if there currently is one and the passed url is nil.

Reply:
Yeah but you explicitly unwrap `state.navigationState.safariOverlays[.articles]` could that ever be nil and then crash the app. I'm not sure if it ever could be, just verifying you considered that

Reply:
ah yeah, I think I was thinking that key lookups in dictionaries return optionals, and this is a dictionary of `[Tab : URL?]`, so I thought this would return a `URL??` and I would force unwrap it to a single `URL?` but not actually 100% sure this is how it works . . .

Lemme playground

Reply:
```swift
//: Playground - noun: a place where people can play

import UIKit

var str = "Hello, playground"

var names: [Int: String?] = [:]

func getOptional(str: String?) -> String {
    if let val = str {
        return val
    } else {
        return "nil!"
    }
}

names[0] = "Ricky"
names[1] = "Bobby"

let bobby = getOptional(str: names[1]!)                     // "Bobby"
let none = getOptional(str: names[2]?.flatMap {return $0})  // "nil!"

// getOptional(str: names[2]!) // CRASH
// getOptional(str: names[0]) // ERROR: Value of type String?? not compatible

```

Reply:
Good catch, it seems like `!` unwraps all the way. I thought it would just strip the outer-most optional.

I think I will use `flatMap` instead

Reply:
I also think I screwed something up with the safari vc delegate, b/c it's not getting called since I moved it into the safari displayer protocol extension.

will investigate tmr . . .

Reply:
addressed. Had to move the safari vc delegate behavior back into the coordinators; something not working about doing protocol conformance implementation from another protocol

Reply:
Here's some more interesting code: 

```swift
var maybeMaybeOne: Int?? = .some(.some(5))
var maybeOne: Int? = maybeMaybeOne!
var one: Int = maybeMaybeOne!!

var maybeMaybeTwo: Int?? = .some(nil)
var maybeTwo: Int? = maybeMaybeTwo!
// var two: Int = maybeMaybeTwo!! CRASH: Unexpectedly found nil
```

Reply:
It seems that ! indeed only strips the outermost ?.

The thing is, a double optional could be `nil`, in which case a force unwrap crashes, or it could be `.some(nil)`, in which case unwrapping once is fine. 

In the dictionary case above, it's going to be `nil` or `.some(.some(URL))`. In hindsight, the dictionary should probably not be to an optional URL but to a normal URL instead